### ISSUE TITLE
Precedence & Periods combined constraint

### ISSUE THREAD

**Issue by ArturMenzeleev** [timnon/pyschedule#57]
@timnon -- first off, this is an extremely useful package. Thank you for making it available. 

I would like to schedule something like 

Task A must occur before Task B
Task B must occur at times [t1,t2,t3,t4]

(The real use case is a bit more complex, but that's the gist of it)

Consider the following minimal example(in the style of test-solvers.py, and using its two_task_scenario()):

```
def PRECEDENCE_PERIODS():
    S = two_task_scenario()
    S['T1'] += S['R1']
    S['T2'] += S['R1']

    S +=S['T2'] > S['T1']
 # S['T2'].periods = [0,1,2,3,4,5,6,7,8,9,10]
    S['T2'].periods = [5,6,7,8,9,10]

    sols = ['[(T1, R1, 1, 2), (T2, R1, 3, 4)]']
    solvers.mip.solve(S, msg=1)
    print(S.solution())
    plotters.matplotlib.plot(S, fig_size=(10, 5), vertical_text=True)

    return S,sols
```

gets me :
```

  File "[REDACTED]Python/2.7/lib/python/site-packages/pyschedule/solvers/mip.py", line 216, in build_mip_from_scenario
    [(x[P.task_right, t_],-1) for t_ in range(t+P.task_left.length+P.offset,self.horizon)]
KeyError: (T2, 1)

```

Allowing any value for T2.periods, or removing the precedence constraint solves the problem .

This should be possible, correct? Or am I going about it wrong?

**Comment by timnon** [comment]
Thanks for this issue, there was a bug, it should be fixed in the new version 0.2.25, please test!

**Comment by ArturMenzeleev** [comment]
Confirming Fix. Apologies for Delay
